### PR TITLE
Feature/fma 47

### PR DESCRIPTION
**Pull request by sksadjad**
contains:
- replacing `onboard` api for `2210vp/participant` with `verify/raw`
- fixed the example for `2210vp/participant/verify/raw`
- replacing `onboard` api for `2210vp/service-offering` with `verify/raw`
- fixed the example for `2210vp/service-offering/verify/raw`

### CODE REVIEW

File: src/methods/common/selfDescription.2210vp.service.ts [hunk 1]
```diff
@@ -72,12 +88,17 @@ export class SelfDescription2210vpService {
       const legalPersonShapeValidation = await this.checkCredentialShape(legalPersonVC, expectedContexts[legalPersonVC.type])
       let serviceOfferingShapeValidation
       if (serviceOfferingVC) {
-        serviceOfferingShapeValidation = this.checkCredentialShape(serviceOfferingVC, expectedContexts[serviceOfferingVC.type])
+        //fixme we're ignoring the shape validation for service-offerings for now, bringg thisback when we have shpaes for v2210vp service-offerings
```
Comment (nklomp) [inline src/methods/common/selfDescription.2210vp.service.ts]:
```suggestion
        //fixme we're ignoring the shape validation for service-offerings for now, bring this back when we have shapes for v2210vp service-offerings
```

File: src/controller/service-offering/service-offering-v2210vp.controller.ts [hunk 2]
```diff
@@ -140,24 +146,16 @@ export class ServiceOfferingV2210vpController {
...
-      serviceOfferingVC as unknown as SignedSelfDescriptionDto<ServiceOfferingSelfDescriptionDto>,
-      get_SD,
-      participant_verif
+      //TODO: fix this later
+      serviceOfferingSelfDescription,
+      {
+        conforms: true,
+        shape: { conforms: true, results: [] },
+        content: { conforms: true, results: [] },
+        isValidSignature: true
```
Comment (nklomp) [inline src/controller/service-offering/service-offering-v2210vp.controller.ts:214]: Can't we validate the signature yet? (we are doing it down below for VCs as well)

File: src/utils/static/.well-known/fullchain.pem [hunk 3]
```diff
@@ -0,0 +1,91 @@
+-----BEGIN CERTIFICATE-----
```
Comment (nklomp) [inline src/utils/static/.well-known/fullchain.pem]: Can this be reverted?

File: src/utils/pipes/ssi-types-parser.pipe.ts [hunk 4]
```diff
@@ -80,12 +69,12 @@ export class SsiTypesParserPipe
   private transformVerifiableCredential(verifiableCredential: VerifiableCredentialDto<any>): TypedVerifiableCredential {
     try {
       const originalVerifiableCredential = { ...verifiableCredential }
-      const type = getTypeFromSelfDescription(verifiableCredential)
-      const { credentialSubject } = verifiableCredential
-      delete verifiableCredential.credentialSubject
+      const type = getTypeFromSelfDescription(originalVerifiableCredential)
+      const { credentialSubject } = originalVerifiableCredential
+      delete originalVerifiableCredential.credentialSubject
```
Comment (nklomp) [inline src/utils/pipes/ssi-types-parser.pipe.ts]: I guess we need to change the name a bit. original suggests it is unaltered, whilst we are manipulating it here

**Review by nklomp — CHANGES_REQUESTED** [review]
Some small remarks

File: src/methods/common/selfDescription.2210vp.service.ts [hunk 5]
```diff
@@ -38,6 +38,22 @@ export class SelfDescription2210vpService {
   ) {}
 
   public async validate(typedVerifiablePresentation: TypedVerifiablePresentation): Promise<validationResultWithoutContent> {
+    let isValidSignature = await this.proofService.validateVP(
```
Comment (sksadjad) [inline src/methods/common/selfDescription.2210vp.service.ts:41]: @nklomp here and in the following lines we're verifying every VC signature that we receive.

File: src/controller/service-offering/service-offering-v2210vp.controller.ts (hunk 2 above)
Comment (sksadjad) [reply src/controller/service-offering/service-offering-v2210vp.controller.ts:214]: here I have mentioned the signature thing. we're verifying all of it. https://github.com/Sphereon/gx-compliance/pull/16/files#r1109997521

File: src/utils/static/.well-known/fullchain.pem (hunk 3 above)
Comment (sksadjad) [reply src/utils/static/.well-known/fullchain.pem]: yes, committed by mistake

File: src/utils/pipes/ssi-types-parser.pipe.ts (hunk 4 above)
Comment (sksadjad) [reply src/utils/pipes/ssi-types-parser.pipe.ts]: yeah, I agree. Made it other way around

**Comment by sksadjad** [comment]
@nklomp can I merge this?

**Comment by nklomp** [comment]
Yes

**Pull request merged**